src/main/java/nl/tudelft/cse1110/andy/writer/standard/StandardResultWriter.java

```diff
@@ -106,6 +110,37 @@ private void printFinalGrade(Context ctx, Result result) {
         }
     }
 
+    private void printZeroGradeReason(Context ctx) {
+        var mas = modeActionSelector(ctx);
+        if (mas == null) {
+            return;
+        }
+
+        StringBuilder reason = new StringBuilder();
+
+        reason.append("\nFinal test score is ");
+
+        if (mas.getMode() == Mode.EXAM) {
+            reason.append("shown as 0/100 during the exam and will be calculated after the exam is over.");
+        } else {
+            reason.append("0/100 ");
+            if (mas.getAction() != Action.FULL_WITH_HINTS && mas.getAction() != Action.FULL_WITHOUT_HINTS) {
+                reason.append("as you are only ");
+                if (mas.getAction() == Action.TESTS)
+                    reason.append("checking if your tests pass");
+                else if (mas.getAction() == Action.META_TEST)
```

Comment:
This should never happen. This is an internal mode used only when we are running the meta tests. What if the method that gets the mode throws an exception in case we get this one? As a sort of post condition.

Reply:
Which method should throw an exception? This one or `getAction()` in `ModeActionSelector`?

Reply:
Done!